x/programs/rust/README.md

```diff
@@ -25,6 +25,100 @@ Just a couple things to note. Serialization is minimal, yet there are certain as
 
 On the Go side, we unmarshal in the same order.
 
+## Storage
+
+A Program has 2 storage types.
+
+- `Memory` in WebAssembly is a linear buffer of unsigned bytes that can read and
+written to by both the guest or host. Because of the very minimal types
+supported by WebAssembly Memory is critical for communication between `Program`
+and host.
```

Comment:
We don't need this last sentence. But if we do, it definitely needs a `,` after `WebAssembly`, then `Memory` should be surrounded by backticks